### ISSUE TITLE
[전 챕터][WikiDocs][Major] 2차 검수 정정 69건이 페이지 재생성으로 유실 · 옛 오류 43건 부활 — 수정이 원천에 반영되지 않음

### ISSUE THREAD

**Issue by modesta-jihye-lee** [yoon-gu/neuqes-101#82]
## 대상

- **챕터**: 전 챕터 (자동 스캔에 걸린 범위: Ch 1·2·4·5·7·8·14·18·19·21·22·25·26·27·28·32)
- **산출물**: WikiDocs (`pages/*.md`) — 다만 **원인은 원천**(`*/*.ipynb`, `book/chapters/*.tex`) 미반영
- **기준 커밋**: `eed3dbe`
- **환경**: 정적 검토 — 커밋 `c94f758` 의 diff 를 현재 파일과 전수 대조(스크립트 첨부). **Ch 01~04 는 Colab 전체 실행 완료** (2026-08-15 · CPU 런타임 · 모두 에러 0건), 나머지 챕터는 미실행

## 실제 결과

2026-06-24 커밋 `c94f758`("WikiDocs 2차 검수: HIGH·MED 사실정확성 수정 (pages 산문)")에서 **실측 대조로 정정한 산문이 되돌아와 있습니다.**

자동 대조 결과: **정정문 69건 유실 · 옛 오류 43건 부활.**

### 원인 — 수정이 하류에만 적용됐습니다

| 커밋 | 일자 | 내용 |
|---|---|---|
| `c94f758` | 2026-06-24 | 2차 검수 정정 — **`pages/` 만** 수정 (커밋 메시지: "코드/▶ 실행 결과 미변경") |
| `3f6f200` | 2026-06-25 | Ch 1–8·10 페이지 **재생성** — 노트북의 옛 서술이 페이지를 덮어씀 |

페이지는 노트북에서 생성되므로, 원천에 옛 서술이 남아 있는 한 **재변환이 돌 때마다 같은 일이 반복**됩니다. 계획서 §8("검수 후 원천 노트북 내용이 바뀌면 WikiDocs·EPUB PASS 를 무효화")의 반대 방향 사고로, 원천이 그대로인 채 하류만 고친 경우입니다.

### 손으로 확인한 대표 사례

| 챕터 | 현재 페이지 (부활한 옛 값) | 2차 검수 정정값 | 확인 근거 |
|---|---|---|---|
| Ch 1 | `datasets` → "Ch 7에서 깊게 본다" | Ch 8 | Ch 8 이 `datasets` 해부 챕터 |
| Ch 1 | `TfidfVectorizer` → "Ch 2-5에서 사용" | Ch 2-6 | Ch 6 도 `TfidfVectorizer()` 사용 |
| Ch 1 | idf 식 `+1` 없음 / "흔할수록 0에 가까워짐" | `+1` 포함 / "1에 가까워짐" | sklearn `smooth_idf=True` |
| Ch 2 | "BERT 는 **Ch 8** 부터 같은 역할" | Ch 9 | README: Ch 9 = DistilBERT 회귀 |
| Ch 4 | "**Ch 11** BERT multi-class" | Ch 12 | Ch 11 = binary softmax |
| Ch 5 | MSE 페널티 "1 vs **16**" | 1 vs 9 | — |
| Ch 8 | "평균 ~150 토큰, `max_length=128` 이 **95% 이상 정상 처리**" | 평균 177 토큰 · 약 51% 절단 | 재실측: **평균 179.6 토큰 / 128 초과 52.0%** |
| Ch 19 | 한국어 WordLevel UNK "**5-15%**" | 약 43.7% | — |
| Ch 21 | "paragraphs **5K**(약 70만-100만 토큰), 약 **3000-5000배** 격차" | 2K(약 27만 토큰), 약 1만배 이상 | 이미 등록된 #70 과 동일 건 |
| Ch 26 | "약 **3M** params(125M 의 40배)" | 약 4.2M(약 30배) | — |

즉 현재 열려 있는 **#70 은 새로 생긴 결함이 아니라 이 되돌림의 결과**입니다.

### 부작용 — 같은 내용이 두 번 실리는 페이지

`pages/01-tfidf-variation.md` 는 재생성으로 돌아온 옛 수식과 `migrate_enrichment.py` 가 되붙인 정정 수식이 **둘 다** 남아, 서로 다른 두 식이 나란히 노출됩니다(별도 이슈로 등록).

## 기대 결과

1. `c94f758` 의 정정 내용을 **원천에 반영** — 노트북 마크다운 셀과 `book/chapters/*.tex` 를 함께 수정
2. 그 뒤 `executed/` → `pages/` 재변환 → EPUB 재빌드로 하류 갱신
3. 앞으로 산문 오류는 **"페이지만 고치기" 를 금지**하고 원천에서 고치는 경로를 기본으로 함
4. 각 챕터 담당 검수자가 아래 스캔 출력의 자기 챕터 항목을 확인해 개별 결함 이슈로 분리 (Ch 1 관련 3건은 등록 완료)

## 재현 절차

```bash
python3 - <<'PY'
import subprocess, collections
diff = subprocess.run(["git","show","c94f758","--unified=0"],capture_output=True,text=True).stdout
res = collections.defaultdict(lambda: {"lost": [], "returned": []}); cur = {}; f = None
for line in diff.split("\n"):
    if line.startswith("+++ b/"): f = line[6:]; continue
    if f is None or not f.startswith("pages/"): continue
    t = line[1:].strip()
    if len(t) < 10: continue
    cur.setdefault(f, open(f).read())
    if line.startswith("+") and not line.startswith("+++") and t not in cur[f]:
        res[f]["lost"].append(t)
    elif line.startswith("-") and not line.startswith("---") and t in cur[f]:
        res[f]["returned"].append(t)
tl = tr = 0
for k in sorted(res):
    l, r = res[k]["lost"], res[k]["returned"]
    if l or r:
        tl += len(l); tr += len(r); print(f"{k}: 유실 {len(l)} / 부활 {len(r)}")
print(f"합계: 유실 {tl} / 부활 {tr}")
PY
```

## 로그

기준 커밋 `eed3dbe` 에서 위 스크립트를 돌린 결과입니다. 챕터 담당자는 자기 챕터 행만 보면 됩니다.

```
pages/01-tfidf-variation.md: 유실 1 / 부활 2
pages/01-tfidf-wrapup.md: 유실 2 / 부활 2
pages/02-sklearn_regression-wrapup.md: 유실 2 / 부활 2
pages/04-softmax_binary-wrapup.md: 유실 1 / 부활 1
pages/05-sklearn_multiclass-anatomy.md: 유실 1 / 부활 1
pages/07-bert_pipeline.md: 유실 1 / 부활 0
pages/08-tokenizer_datasets-wrapup.md: 유실 1 / 부활 1
pages/08-tokenizer_datasets.md: 유실 1 / 부활 0
pages/14-auxiliary_loss.md: 유실 13 / 부활 1
pages/18-ko_auxiliary-wrapup.md: 유실 1 / 부활 1
pages/19-tokenizer_training-wrapup.md: 유실 1 / 부활 1
pages/19-tokenizer_training.md: 유실 1 / 부활 0
pages/21-en_bert_classify.md: 유실 6 / 부활 3
pages/22-ko_bert_pretrain-wrapup.md: 유실 1 / 부활 1
pages/25-gpt2_continual_pretrain-wrapup.md: 유실 1 / 부활 1
pages/26-ko_tiny_gpt-wrapup.md: 유실 2 / 부활 2
pages/26-ko_tiny_gpt.md: 유실 7 / 부활 4
pages/27-ko_gpt2_continual_pretrain-wrapup.md: 유실 1 / 부활 1
pages/27-ko_gpt2_continual_pretrain.md: 유실 6 / 부활 5
pages/28-sft.md: 유실 2 / 부활 0
pages/32-diffusion_intro-wrapup.md: 유실 11 / 부활 12
pages/32-diffusion_intro.md: 유실 6 / 부활 2
합계: 유실 69 / 부활 43
```

건수가 몰린 곳은 `pages/14-auxiliary_loss.md`(유실 13)·`pages/32-diffusion_intro-wrapup.md`(유실 11 / 부활 12)·`pages/26-ko_tiny_gpt.md`(유실 7)로, 담당 페어는 B·D·C 입니다.

## 추가 확인

- "유실" 로 집계된 항목에는 **이후 정당하게 다시 쓰인 문장**이 섞일 수 있습니다. 위 표의 10건은 현재 파일·실측과 직접 대조해 확인한 것이고, 나머지는 챕터 담당자의 확인이 필요합니다.
- 이 스캔은 `c94f758` 한 커밋만 기준으로 합니다. 다른 검수 수정 커밋에도 같은 일이 있었는지는 별도 확인이 필요합니다.

## 수정 범위 원칙 · 재실행 여부

가이드 3쪽에 따라 WikiDocs 검수에서 고치는 것은 산문뿐이고 코드·▶ 실행 결과는 `executed/` 가 단일 출처입니다. 이 건은 전부 산문이지만, **원천(노트북 마크다운 셀·`book/chapters/*.tex`)이 옛 서술을 그대로 갖고 있어 페이지만 고치면 다음 재변환에서 되돌아옵니다.**

- **노트북 재실행 불필요** — 마크다운 셀 수정이므로 코드·출력은 변하지 않습니다.
- WikiDocs 재변환 → EPUB 재빌드는 필요합니다. 해당 챕터의 WikiDocs·EPUB PASS 는 계획서 §8 대로 무효화됩니다.
- 재검수자는 **재변환을 한 번 돌린 뒤 위 스캔을 다시 실행해 합계가 0 이 되는지**로 확인합니다. 페이지에서만 고치면 이 수치가 다시 올라옵니다.

## 심각도

**Major** — 이미 한 번 검수해 고친 사실오류가 출판 경로 산출물에 되살아나 있습니다. 조치하지 않으면 재변환마다 반복되어 개별 챕터의 PASS 를 신뢰할 수 없게 됩니다.

## 담당

- 수정 담당자: 미정
- 재검수자: 등록자 (@modesta-jihye-lee)
- 연결: #70, 챕터 검수 이슈 #25, 검수 기준 논의 #63

**Comment by fluentmin** [comment]
@modesta-jihye-lee 확인 감사드립니다. 말씀해주신 대로 검수 시 산문 오류는 "페이지(`pages` 하위 md 파일들)만 고치기" 를 금지하고 원천 노트북 파일을 고치는 것을 원칙으로 삼는 게 어떠하신가요?

**Comment by yum-yeom** [comment]
Ch 22 개별 사례 2곳 확인 — #190 등록, PR #191 로 수정 (practice "약 500-1,500 블록" 옛 문단 · anatomy §7-2 옛 도입 문단). 메커니즘: 재변환 시 migrate_enrichment.py 가 직전 git 판의 보강 산문을 이식해 노트북 정정과 병존.
